lib/Serialization/DeserializeSIL.cpp

```diff
@@ -379,6 +379,95 @@ SILType SILDeserializer::getSILType(Type Ty, SILValueCategory Category,
       .getCategoryType(Category);
 }
 
+unsigned SILDeserializer::readNextRecord(SmallVectorImpl<uint64_t> &scratch) {
+  scratch.clear();
+  unsigned kind = 0;
+  llvm::Expected<llvm::BitstreamEntry> maybeEntry =
+      SILCursor.advance(AF_DontPopBlockAtEnd);
+  if (!maybeEntry)
+    MF->fatal(maybeEntry.takeError());
```

Comment:
I don't know if the rest of the deserializer also works this way, but I hope this doesn't call `exit()`. The compiler is also embedded in some long-lived processes (for example lldb) and terminating the process is not a great experience there. If possible returning (and bubbling up) an `llvm::Expected<unsigned>` is much better.

Reply:
I have bubbled up the error. Calling fatal seems to be a pretty common pattern in the deserializer.